### PR TITLE
backstage: remedy o header DAC issues

### LINKED ISSUE #901: o-header: inconsistent label for navigation menu
The menu button provided for users in the header, has been labelled as ‘open side navigation menu’. Once the menu is opened, the close button is labelled as ‘Close drawer menu’. This can be confusing for users of screen reading assistive technologies, as they are not informed that both components refer to the same menu. This can be increasingly difficult to discern when navigating the page out of context using the forms list using the names of forms alone.

*for this to be classed as done, we need to update origami demos and work with the platforms team to update pagekit*

https://docs.google.com/document/d/17-d7DAAORFCaUbiHdFPqvrBerUKbMyhG/edit#bookmark=id.9ropi7us8epm

image-20221012-144826

### PR DESCRIPTION
**Pull request by akomiqaia**
update o header close button label and title so it matches open menu label; closes #901

### CODE REVIEW

File: components/o-header/demos/src/header.mustache [hunk 1]
```diff
@@ -171,37 +171,37 @@
...
-							{{/hasChildren}}
-						</li>
-						{{/items}}
-				{{/nav}}
-			</ul>
-		</nav>
+		{{#nav}}
+			<nav class="o-header__drawer-menu o-header__drawer-menu--primary">
+				<ul class="o-header__drawer-menu-list" {{#heading}}aria-label="{{heading.name}}"{{/heading}}>
+							{{#heading}}
```
Comment (notlee) [inline components/o-header/demos/src/header.mustache:177]: Breaking this menu up into multiple smaller menus seems sensible!

1. Where did this come from? It's not part of https://github.com/Financial-Times/origami/issues/901?
2. Would it make more sense to move the `aria-label` to the `nav` element instead, for users navigating by landmark?


Screenshot 2022-11-30 at 17 05 45

**Comment by notlee** [comment]
We're going to need to update page-kit (or ask the platform team to) so we can rollout these changes to ft.com. We'll also want to update the [navigation service example](https://www.ft.com/__origami/service/navigation/v2/docs/example), which some people copy markup from. I'll create an issue for that before merging this :)

**Comment by notlee** [comment]
https://github.com/Financial-Times/origami/issues/907

File: components/o-header/demos/src/header.mustache (hunk 1 above)
Comment (akomiqaia) [reply components/o-header/demos/src/header.mustache:177]: not sure what you mean by the first question. I also worked on [this issue](https://github.com/Financial-Times/origami/issues/902).

for the second one, you are right. We should move it to the `nav` tag.

Comment (akomiqaia) [reply components/o-header/demos/src/header.mustache:177]: one thing is that when I move `aria-label` to nav tag VoiceOver did not announce there was a list of items. but if we keep the `aria-label` on the `ul` tag then the voiceover reads the heading and announces that there are 11 (for example) items in this list.

Comment (notlee) [reply components/o-header/demos/src/header.mustache:177]: Good point. Taking another look at this I see them as all one `nav`, but with multiple sections within that nav. What do you think about going back to one `nav` element for these and but splitting into multiple `ul` lists with a `aria-label`?

Comment (notlee) [reply components/o-header/demos/src/header.mustache:177]: In fact, forget that.

The first list item isn't really an item of the nav – it's not a link, and more of a heading – so it shouldn't be in the list at all. Looks like we should use a `h2` heading level. DAC recommended this in their report and there's also a [WAI guide to navigation here](https://www.w3.org/WAI/tutorials/menus/structure/#label-menus) which does the same.

Maybe something like this?

```html
<nav>
	<h2 id="o-header-drawer-top-sections">Top sections</h2>
	<ul aria-labelledby="o-header-drawer-top-sections">
	    <!-- list items -->    
	</ul>
	<h2 id="o-header-drawer-top-sections">FT recommends</h2>
	<ul aria-labelledby="o-header-drawer-top-sections">
	    <!-- list items -->    
	</ul>
</nav>
<nav>
	    <!-- secondary stuff -->    
</nav>
<nav>
	    <!-- user stuff -->    
</nav>
```
drawer-markup

Comment (notlee) [reply components/o-header/demos/src/header.mustache]: I've created a new issue to capture this:
https://github.com/Financial-Times/origami/issues/910

The [previous issue](https://github.com/Financial-Times/origami/issues/902) was about something else, but included the wrong description. 🙈

**Comment by notlee** [comment]
👍 Merging to the WIP v11 branch. Solves [inconsistent labels](https://github.com/Financial-Times/origami/issues/910), more work is required to markup [visual navigation headings](https://github.com/Financial-Times/origami/issues/910)

**Pull request merged**